desc/objectives/utils.py

```diff
@@ -86,6 +83,21 @@ def get_fixed_boundary_constraints(
     return constraints
 
 
+def maybe_add_self_consistency(constraints):
+    """Add self consistency constraints if needed."""
```

Comment:
"If needed" just means if the constraints do not currently contain R consistency, Z consistency, or FixLambdaGauge, it will add them to the constraints? So It will almost have to add these?

Reply:
Yeah basically. The assumption is that the user doesn't need to specify those constraints explicitly. In the same vein, I currently have it so the `get_fixed_boundary_constraints` etc don't return the self consistency constraints, but we can change that if we want. My thinking was that the stuff returned by `get_fixed...` is stuff the user may take as a guide for custom constraints and modifying stuff, but we don't want them to be messing with the self consistency stuff.